### PR TITLE
[WIP] Update libtest to version 0.0.2

### PR DESCRIPTION
**Pull request by gnzlbg**
This updates libtest to version 0.0.2.

r? @alexcrichton

### CODE REVIEW

File: src/libtest/Cargo.toml [hunk 1]
```diff
@@ -7,10 +7,10 @@ edition = "2018"
 [lib]
 name = "test"
 path = "lib.rs"
-crate-type = ["dylib", "rlib"]
+crate-type = ["rlib"]
```
Comment (gnzlbg) [inline src/libtest/Cargo.toml]: I suppose I can just remove this line.

**Comment by gnzlbg** [comment]
This is blocked on #59336 landing on nightly.

**Comment by Centril** [comment]
https://github.com/rust-lang/rust/pull/59336 has landed.

**Comment by gnzlbg** [comment]
Now the build gets up to the ui tests, but some tests fail with the error:

```
error[E0658]: use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)
```

It's unclear which of the sysroot crates is the problematic one.

**Comment by xales** [comment]
> Now the build gets up to the ui tests, but some tests fail with the error:
> 
> ```
> error[E0658]: use of unstable library feature 'rustc_private': this crate is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)
> ```
> It's unclear which of the sysroot crates is the problematic one.

Not sure if this makes any sense what I did, but in looking into what's going one here:
I hacked up that message and had it put in the crate name being marked with rustc_private. Most of the errors reported `libtest` as the crate its hitting the `rustc_private` on. One or two hit `libc` and at least one, the `crate_name` query returned the empty string... not sure if that helps!

File: src/test/ui-fulldeps/hash-stable-is-unstable.stderr [hunk 2]
```diff
@@ -2,47 +2,55 @@ error[E0601]: `main` function not found in crate `hash_stable_is_unstable`
...
 LL | #[derive(HashStable)]
    |          ^^^^^^^^^^
    |
    = help: add #![feature(rustc_private)] to the crate attributes to enable
 
-error: aborting due to 6 previous errors
+error[E0658]: use of unstable library feature 'rustc_private': crate "rustc" is being loaded from the sysroot, an unstable location; did you mean to load this crate from crates.io via `Cargo.toml` instead? (see issue #27812)
+  --> $DIR/hash-stable-is-unstable.rs:13:10
+   |
+LL | #[derive(HashStable)]
```
Comment (oli-obk) [inline src/test/ui-fulldeps/hash-stable-is-unstable.stderr]: You need to duplicate the annotation in https://github.com/rust-lang/rust/blob/d8b0f751908b8b03a61ccfe08085e7e77d984a5f/src/test/ui-fulldeps/hash-stable-is-unstable.rs#L14 by adding

```rust
//~| use of unstable library feature 'rustc_private'
```

below

**Comment by Dylan-DPC** [comment]
ping from triage @alexcrichton waiting for your review @gnzlbg you have conflicts to resolve

**Comment by alexcrichton** [comment]
This has my approval, it's waiting for tests to pass

**Comment by jonas-schievink** [comment]
Triage ping @gnzlbg, this is waiting on merge conflicts to be resolved, and the tests need to pass

**Comment by JohnCSimon** [comment]
Howdy! This is a Triage ping for @gnzlbg,  
This is waiting on merge conflicts to be resolved, 
The tests need to pass.

**Comment by JohnCSimon** [comment]
@gnzlbg Triage again - I'm closing this one due to to inactivity 
Thanks

**Comment by JohnCSimon** [comment]
@gnzlbg @Centril 
Hi! this PR hasn't seen much movement in a week. I'll be closing this next week.

**Comment by JohnCSimon** [comment]
@gnzlbg Unfortunately this has sat idle for far too long, and I'm going to have to close this. Thanks for contribution anyway.

**Pull request closed without merging**